app/data_client.go

```diff
@@ -430,12 +517,12 @@ func (d *DataClient) TabularDataByFilter(
 	}
 	// TabularData contains tabular data and associated metadata
 	dataArray := []TabularData{}
-	var metadata *pb.CaptureMetadata
 	for _, data := range resp.Data {
-		if len(resp.Metadata) > 0 && int(data.MetadataIndex) < len(resp.Metadata) {
+		var metadata *pb.CaptureMetadata
+		switch {
```

Comment:
(q) curious why the switch (heh) to a `switch` statement here? They're functionally basically identical so no need to change back if you're happy with it, but I typically think of `if/else` as more idiomatic when checking boolean values, especially a single value.

Reply:
Actually I suspect you can just get rid of the default/else case entirely by instantiating `metadata` as a concrete object rather than a pointer. It'll default to the empty case anyway, then we update it when it makes sense to do so.

Reply:
oh interesting! I'm surprised the linter complained but good to know!

Reply:
also, I forgot I set it as a pointer because when we don't we get this Mutex.Lock error:
```
app/data_client.go:523:15: copylocks: assignment copies lock value to metadata: go.viam.com/api/app/data/v1.CaptureMetadata contains google.golang.org/protobuf/runtime/protoimpl.MessageState contains sync.Mutex (govet)
			metadata = *resp.Metadata[data.MetadataIndex]
			           ^
make: *** [Makefile:67: lint-go] Error 1
```
so if it is okay with you, I am going to revert back to the pointer w/ an empty default. In this case if our metadata index is out of bounds it will just give us back empty metadata

Reply:
Ahh yeah, that's totally reasonable. I didn't realize the mutex issue but that definitely makes sense!